modules/internal/DefaultRectangular.chpl

```diff
@@ -1769,7 +1769,7 @@ module DefaultRectangular {
     }
 
     proc dsiTargetLocales() {
-      compilerError("targetLocales is unsupported by default domains");
+      return [this.data.locale, ];
```

Comment:
Yes, this is what I was imagining.  I think it needs to be added to the the domain as well (?).

As extra credit, I also ultimately think that we should not require dsiTargetLocales on both array and domain objects, but should support it on domains only and have arrays answer the target locales query by passing it along to their domains...